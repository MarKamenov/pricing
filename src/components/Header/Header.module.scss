.header {
  display: flex;
  justify-content: flex-end;
  padding: var(--spacing-xs) var(--spacing-sm) var(--spacing-xs);
  background-color: var(--header-bck);

  @media (min-width: 768px) {
    padding-left: var(--spacing-xl);
    padding-right: var(--spacing-xl);
  }
}