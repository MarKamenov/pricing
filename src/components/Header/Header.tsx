import { Moon, Sun } from 'lucide-react';

import styles from './Header.module.scss';
import { useTheme } from '../../contexts';
import { Button } from '../Button';

export const Header = () => {
  const { theme, toggleTheme } = useTheme();

  return (
    <header className={styles.header}>
      <Button variant='secondary'
        onClick={toggleTheme}
        ariaLabel={`Switch to ${theme === 'light' ? 'dark' : 'light'} theme`}
      >
        {theme === 'light' ? <Moon /> : <Sun />}
      </Button>
    </header>
  );
};