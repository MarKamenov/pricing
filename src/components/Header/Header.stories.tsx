import type { Meta, StoryObj } from '@storybook/react';
import { Header } from './Header';
import { ThemeProvider } from '../../contexts';

const meta: Meta<typeof Header> = {
    title: 'Components/Header',
    component: Header,
    parameters: {
        layout: 'fullscreen',
    },
    decorators: [
        (Story) => (
            <ThemeProvider>
                <Story />
            </ThemeProvider>
        ),
    ],
    tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof Header>;

export const HeaderComponent: Story = {
    decorators: [
        (Story) => (
            <ThemeProvider>
                <Story />
            </ThemeProvider>
        ),
    ],
};

