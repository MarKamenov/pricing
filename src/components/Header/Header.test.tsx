import { render } from '@testing-library/react';
import { Header } from './Header';

jest.mock("../../contexts/ThemeContext", () => ({
    useTheme: () => ({
        theme: "dark",
        toggleTheme: jest.fn(),
    }),
}));

describe('Header component', () => {

    test('should render component', () => {
        const { container } = render(<Header />)

        expect(container).toBeDefined();
    });

    test('should match snapshot', () => {
        const { container } = render(<Header />);

        expect(container).toMatchSnapshot();
    });
});

