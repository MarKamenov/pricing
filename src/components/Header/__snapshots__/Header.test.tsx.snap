// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <button
      aria-label="Switch to light theme"
      class="button secondary small"
    >
      <svg
        class="lucide lucide-sun"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="4"
        />
        <path
          d="M12 2v2"
        />
        <path
          d="M12 20v2"
        />
        <path
          d="m4.93 4.93 1.41 1.41"
        />
        <path
          d="m17.66 17.66 1.41 1.41"
        />
        <path
          d="M2 12h2"
        />
        <path
          d="M20 12h2"
        />
        <path
          d="m6.34 17.66-1.41 1.41"
        />
        <path
          d="m19.07 4.93-1.41 1.41"
        />
      </svg>
    </button>
  </header>
</div>
`;

exports[`Title component should match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <button
      aria-label="Switch to light theme"
      class="button secondary small"
    >
      <svg
        class="lucide lucide-sun"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="4"
        />
        <path
          d="M12 2v2"
        />
        <path
          d="M12 20v2"
        />
        <path
          d="m4.93 4.93 1.41 1.41"
        />
        <path
          d="m17.66 17.66 1.41 1.41"
        />
        <path
          d="M2 12h2"
        />
        <path
          d="M20 12h2"
        />
        <path
          d="m6.34 17.66-1.41 1.41"
        />
        <path
          d="m19.07 4.93-1.41 1.41"
        />
      </svg>
    </button>
  </header>
</div>
`;
